nodedb.go

```diff
@@ -98,18 +93,14 @@ func newNodeDB(db dbm.DB, cacheSize int, opts *Options) *nodeDB {
 	}
 
 	return &nodeDB{
-		db:                 db,
-		batch:              db.NewBatch(),
-		opts:               *opts,
-		latestVersion:      0, // initially invalid
-		nodeCache:          make(map[string]*list.Element),
-		nodeCacheSize:      cacheSize,
-		nodeCacheQueue:     list.New(),
-		fastNodeCache:      make(map[string]*list.Element),
-		fastNodeCacheSize:  100000,
-		fastNodeCacheQueue: list.New(),
-		versionReaders:     make(map[int64]uint32, 8),
-		storageVersion:     string(storeVersion),
+		db:             db,
+		batch:          db.NewBatch(),
+		opts:           *opts,
+		latestVersion:  0, // initially invalid
+		nodeCache:      cache.New(cacheSize),
+		fastNodeCache:  cache.New(fastNodeCacheLimit),
```

Comment:
why do we still need `fastNodeCache`?

Reply:
According to benchmarks, removing it significantly degrades the performance: https://github.com/osmosis-labs/osmosis/issues/1041

Reply:
I looked at it too fast. I don't see a benchmark with `nodeCache and no-fastNodeCache`. The benchmark you have is `noCache` vs `with fastNodeCache`. Sorry, maybe I'm not interpreting it correctly as I don't see the whole setting.

Reply:
I think I just wasn't consistent with how I was referring to the benchmarks, sorry for the confusion.

The comparison is most certainly between `both node caches` and `regular cache but no-fastNodeCache`.
When I referred to:
`cache.log` = `both node caches`
`nocache.log` = `regular cache but no-fastNodeCache`

Looking [at the benchstat](https://github.com/osmosis-labs/osmosis/issues/1041#issuecomment-1058788599), the delta is coming mostly from `query-hits-fast-16` and `iteration-fast-16`, proving that the regular cache was untouched. If it was touched, we would see delta in `query-hits-slow-16` and `iteration-slow-16 `

To explain the reasons for trying this, our node operators were getting OOM'ed and pprof results were showing memory growth only specific to fast cache. There was never a motivation for trying things w/o regular cache.

To mitigate the memory growth we decided to hardcode the value for the fast cache to `100000`
https://github.com/cosmos/iavl/blob/d07ef28ff2421163d95092fa4b4d27c41080de2f/nodedb.go#L33

The value was chosen by spinning up a bunch of nodes at 10k, 50k, 100k, 150 and 200k. 100k proved to be the most optimal.

Spinning up several nodes to find a good value seemed like an unnecessarily complex solution. That's why I decided to do this refactor to eventually introduce byte cache for fast nodes. I'm also hoping to expose a parameter in config to let node operators choose this byte value.